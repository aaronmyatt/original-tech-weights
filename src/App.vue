<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>

<template>
<div id="app">
  <section class="container-fluid">

    <Brand :brandName="getBrandName()" />
    <img src="./assets/logo.png">
    <UserDetails v-bind="getUserDetails()" />

  </section>

  <br/>

  <section class="container-fluid">

    <ProjectList :projectList="getProjectList()" />

  </section>


</div>
</template>

<script>

import projectDetails from './data/ProjectDetails';
import Brand from './components/Brand';
import UserDetails from './components/UserDetails';
import ProjectList from './components/ProjectList';

export default {
  name: 'app',
  components: {
    Brand,
    UserDetails,
    ProjectList,
  },
  methods: {
    getBrandName() {
      return projectDetails.projectDetails().brandName;
    },
    getUserDetails() {
      return projectDetails.projectDetails().userDetails;
    },
    getProjectList() {
      return projectDetails.projectDetails().projectList;
    },
  },
};
</script>
