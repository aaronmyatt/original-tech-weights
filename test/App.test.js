import { shallow } from 'vue-test-utils'
import App from '../src/App'

describe('App', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(App)
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})
