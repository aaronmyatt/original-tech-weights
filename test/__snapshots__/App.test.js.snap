// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has the expected html structure 1`] = `
<div
  id="app"
>
  <section
    class="container-fluid"
  >
    <!---->
     
    <img
      src="./assets/logo.png"
    />
     
    <!---->
  </section>
   
  <br />
   
  <section
    class="container-fluid"
  >
    <!---->
  </section>
</div>
`;
